src/python/pants/testutil/engine/util.py

```diff
@@ -197,3 +199,10 @@ def green(self, text):
 
   def red(self, text):
     return self._safe_color(text, red)
+
+
+def mock_goal_options(goal: Type[Goal], scoped_options: Dict[str, Any]) -> Goal.Options:
```

Comment:
Goal.Options is an (untyped) function, this returns the return type of Goal.Options (which is not a function). Type checking is surely wrong here.

Reply:
@jsirois : It's a `@classproperty`... I'm not sure whether MyPy allows for checking that kind of construction, but I could imagine it being roughly equivalent to something like:
```
class Outer: ..

class Goal:
  Inner = Outer
```
... so not too far-fetched that it would work. But yea: I apologize for this construction.

Reply:
Gotcha. This is all pretty evil / magic it works at all. Basically, annotations can have side-effects and the @classproperty is called with side-effect of returning the real type:
```python
$ python
Python 3.8.0 (default, Oct 23 2019, 18:51:26) 
[GCC 9.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def Annotation():
...   print("Bad magic.")
...   return int
... 
>>> def annotated() -> Annotation():
...   return 42
... 
Bad magic.
>>> import inspect
>>> sig = inspect.signature(annotated)
>>> sig.return_annotation
<class 'int'>
>>> 
```

Reply:
I completely agree... I put a large comment on it to apologize.